src/ParentComponent.js

```diff
@@ -0,0 +1,22 @@
+import PureComponent from "./PureComponent";
+import ClassComponent from "./ClassComponent";
+import React from "react";
```

Comment:
I would suggest to use destructure here 
```suggestion
import { Component } from "react";
```

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment#object_destructuring

Reply:
with this change i am getting error stating 
TypeError: Cannot read properties of undefined (reading 'Component')
not sure why is this happening